lib/column/codegen/column.tpl

```diff
@@ -378,10 +381,14 @@ func (col *{{ .ChType }}) AppendRow(v interface{}) error {
         col.col.Append(val)
 	{{- end }}
 	default:
-		return &ColumnConverterError{
-			Op:   "AppendRow",
-			To:   "{{ .ChType }}",
-			From: fmt.Sprintf("%T", v),
+		if rv := reflect.ValueOf(v); rv.Kind() == col.ScanType().Kind() && rv.CanConvert(col.ScanType()) {
+			col.col.Append(rv.Convert(col.ScanType()).Interface().({{ .GoType }}))
+		} else {
+			return &ColumnConverterError{
+				Op:   "AppendRow",
+				To:   "{{ .ChType }}",
+				From: fmt.Sprintf("%T", v),
+			}
```

Comment:
Supporting base types:
I had to add additional check
```
rv.Kind() == col.ScanType().Kind()
```

because you have one test (SimpleInt) that should fail (int should not be converted into int64)

so how it works:

when we have type int64 and custom int64 type:
```
type customInt64 int64
```

var c customInt64
reflect.TypeOf(c).Kind() == reflect.Int64

that's why it works well

from documentation:

```
The second property is that the Kind of a reflection object describes the underlying type, not the static type. If a reflection object contains a value of a user-defined integer type, as in

type MyInt int
var x MyInt = 7
v := reflect.ValueOf(x)
the Kind of v is still reflect.Int, even though the static type of x is MyInt, not int. In other words, the Kind cannot discriminate an int from a MyInt even though the Type can.
```

https://go.dev/blog/laws-of-reflection

Reply:
nice